# Chinook Database SQL Analysis

This project analyzes sales and customer data from the Chinook music store database using SQL.

**Objectives:**

- Analyze revenue performance
- Identify top customers and products
- Examine sales trends

Database: SQLite  
Queries: Stored in `/queries` folder

In [17]:
import sqlite3
import pandas as pd

# Connect to the database
conn = sqlite3.connect('../data/Chinook_Sqlite.sqlite')

---

## Question 1: What is the total revenue?

**Business Question:**

Understanding total revenue provides a high-level view of overall business performance.

In [40]:
# Question 1 — Total Revenue

with open('../queries/01_total_revenue.sql', 'r') as file:
    query = file.read()

total_revenue = pd.read_sql(query, conn)

total_revenue

Revenue
0   2328.6

**Insight:**

The Chinook store has generated a total revenue of **$2328.60**.

This metric serves as a baseline for evaluating customer value, product performance, and sales trends in subsequent analysis.

---

## Question 2: How many customers does the store have?

**Business Question:**

Understanding the number of unique customers purchasing from the store gives us important descriptive information.

In [19]:
# Question 2 — Total Customers

with open('../queries/02_total_customers.sql', 'r') as file:
    query = file.read()

total_customers = pd.read_sql(query, conn)

total_customers

COUNT(DISTINCT CustomerId)
0                          59

**Insight:**

The Chinook store has **59** unique customers. 

This indicates the store operates with a relatively small, defined customer base.

This information could be useful for customer retention, loyalty programs, and more generally to understand customer demographics.

---

## Question 3: Which countries generate the most revenue?

**Business Question:**

Exploring which countries generate the most revenue can give insight into market strength in different locations.

In [71]:
# Question 3 - Revenue by country

with open('../queries/03_country_revenue.sql', 'r') as file:
    query = file.read()

country_revenue = pd.read_sql(query, conn)

country_revenue

BillingCountry  Revenue
0             USA   523.06
1          Canada   303.96
2          France   195.10
3          Brazil   190.10
4         Germany   156.48
5  United Kingdom   112.86
6  Czech Republic    90.24
7        Portugal    77.24
8           India    75.26
9           Chile    46.62

**Insight:**

The Chinook store serves **24** different geographical regions. The largest markets are in **USA, Canada, and France**.

Based on revenue, many of these countries only have one or two customers. 

Marketing efforts should continue to prioritize the largest market, USA, while exploring growth opportunities in smaller markets.

---

## Question 4: Who are the top 10 customer by total spending?

**Business Question:**

Identifying top customers can be useful for loyalty and/or reward programs.

In [45]:
# Question 4 - Top 10 customers

with open('../queries/04_top_customers.sql', 'r') as file:
    query = file.read()

top_customers = pd.read_sql(query, conn)

display(top_customers)

top_10_revenue = top_customers['CustomerSpend'].sum()
top_10_pct = (top_10_revenue / total_revenue['Revenue'].iloc[0]) * 100
print(f"Top 10 customers account for {top_10_pct:.2f}% of total revenue")

CustomerId                Name  CustomerSpend         Country
0           6         Helena Holý          49.62  Czech Republic
1          26  Richard Cunningham          47.62             USA
2          57          Luis Rojas          46.62           Chile
3          45     Ladislav Kovács          45.62         Hungary
4          46       Hugh O'Reilly          45.62         Ireland
5          24       Frank Ralston          43.62             USA
6          28       Julia Barnett          43.62             USA
7          37     Fynn Zimmermann          43.62         Germany
8           7       Astrid Gruber          42.62         Austria
9          25      Victor Stevens          42.62             USA

Top 10 customers account for 19.38% of total revenue


**Insight:**

The top 10 customers account for 19.38% of total revenue. This indicates that spending is fairly evenly distributed.

This group of customers spent between **$42.62** and **$49.62**, with **Helena Holý, Richard Cunningham, and Luis Rojas** as the top 3 spenders. 

These high-value customers represent a key target for loyalty programs, personalised promotions, or retention initiatives.  

---

## Question 5: What are the top 10 tracks by total revenue?

**Business Question:**

Identifying top songs by revenue can provide insight into what customers are listening to and why.

In [ ]:
# Question 5 - Top 10 Tracks

with open('../queries/05_top_tracks.sql', 'r') as file:
    query = file.read()

top_tracks = pd.read_sql(query, conn)

top_tracks

TrackId                          Name                Artist  \
0     2832                The Woman King  Battlestar Galactica   
1     2850                       The Fix                Heroes   
2     2868                     Walkabout                  Lost   
3     3177                      Hot Girl            The Office   
4     3200                Gay Witch Hunt            The Office   
5     3214             Phyllis's Wedding            The Office   
6     3223  How to Stop an Exploding Man                Heroes   
7     3250                         Pilot               Aquaman   
8     2820        Occupation / Precipice  Battlestar Galactica   
9     2821                 Exodus, Pt. 1  Battlestar Galactica   

                            Album  Revenue  
0  Battlestar Galactica, Season 3     3.98  
1                Heroes, Season 1     3.98  
2                  Lost, Season 1     3.98  
3            The Office, Season 1     3.98  
4            The Office, Season 3     3.98  
5            The Office, Season 3     3.98  
6                Heroes, Season 1     3.98  
7                         Aquaman     3.98  
8  Battlestar Galactica, Season 3     1.99  
9  Battlestar Galactica, Season 3     1.99

**Insight:**

The top 10 songs by revenue are shared between 3 TV series. Customers are buying soundtracks from:
- **Battlestar Galactica, Season 3**
- **The Office Season 1/3**
- **Heroes, Season 1**

The distribution of track revenue is mostly flat. No single track is dominating sales. The highest revenue from a single track is only **$3.98**, which means it was only purchased twice. 

Customers are mostly buying individual tracks and their listening preferences are not overlapping. 

---

## Question 6: Which music genres generate the most revenue?

**Business Question:**

Looking at which genres are generating the most revenue can inform what the store stock going forward. 

In [70]:

with open('../queries/06_top_genres.sql', 'r') as file:
    query = file.read()

top_genres = pd.read_sql(query, conn)

display(top_genres)

top_5_genres = top_genres['Revenue'].sum()
top_5_pct = (top_5_genres / total_revenue['Revenue'].iloc[0]) * 100
print(f"Top 5 genres account for {top_5_pct:.2f}% of total revenue")

Genre  Revenue
0                Rock   826.65
1               Latin   382.14
2               Metal   261.36
3  Alternative & Punk   241.56
4            TV Shows    93.53

Top 5 genres account for 77.52% of total revenue


**Insight**

**Rock** has the highest genre revenue at **$826.65**, followed by **Latin ($382.14)**, and **Metal ($261.36)**. The top 5 genres account for **77.52%** of total sales.

This information can guide marketing and merchandising decisions by highlighting popular genres. 

Customers purchasing tracks within these genres could be shown similar popular tracks within the genre to encourage additional sales. 